NGCHM/WebContent/javascript/API.js

```diff
@@ -54,6 +55,17 @@
 	return heatMap.getSummaryHist (heatMap.getCurrentDL(), +thresholds[0], +thresholds[thresholds.length-1]);
     }
 
+    // Return a promise for an array of all linkout types available for the NG-CHM.
+    // The promise resolves after the NG-CHM's customization file has loaded.
+    //
+    function getLinkoutTypes () {
+	return new Promise ((resolve) => {
+	    NgChm.CUST.waitForPlugins (() => {
+		resolve (NgChm.CUST.linkoutTypes.slice(0));
```

Comment:
Unless these "linkoutTypes" are supposed to be modifiable, this would be better if it returned a deep-copy.  More so when this function is meant for external use.